### PR TITLE
post-voa channel power, rx low power warning and alarm thresholds, and module general status

### PR DESCRIPTION
**Pull request by samisaab**

### CODE REVIEW

File: inc/tainetworkif.h [hunk 1]
```diff
@@ -511,6 +511,30 @@ typedef enum _tai_network_interface_attr_t
...
+     * @type #tai_float_t
+     * @flags READ_ONLY
+     */
+    TAI_NETWORK_INTERFACE_ATTR_CURRENT_POST_VOA_PROVISIONED_CHANNEL_POWER,
+
+    /**
+     * @brief The RX power low warning threshold in dBm
+     *
+     * @type #tai_float_t
+     * @flags READ_ONLY
```
Comment (ishidawataru) [inline inc/tainetworkif.h]: @samisaab Can't we change this threshold?

How do we know when the rx power fall below this threshold?
Do we need to add an attribute to get an alarm for this?

Comment (samisaab) [reply inc/tainetworkif.h]: Yes.  I'll take a look at implementing the alarm and let you know if I have questions.

File: inc/tainetworkif.h [hunk 2]
```diff
@@ -511,6 +511,30 @@ typedef enum _tai_network_interface_attr_t
      */
     TAI_NETWORK_INTERFACE_ATTR_NOTIFY,
 
+    /**
+     * @brief The current post-VOA RX input power in the provisioned channel in dBm
+     *
+     * @type #tai_float_t
+     * @flags READ_ONLY
+     */
+    TAI_NETWORK_INTERFACE_ATTR_CURRENT_POST_VOA_PROVISIONED_CHANNEL_POWER,
```
Comment (JumpeiHongo) [inline inc/tainetworkif.h]: @samisaab 
What is the register name for ACO?
I only know "Current Post-VOA Total Power".

Comment (samisaab) [reply inc/tainetworkif.h]: I'm currently using the four RX power monitoring attributes as follows.

TAI_NETWORK_INTERFACE_ATTR_CURRENT_INPUT_POWER
    Pre-VOA Total Power
    OIF register 0xB4E0: Current Input Power [Total Rx Optical]

TAI_NETWORK_INTERFACE_ATTR_CURRENT_POST_VOA_TOTAL_POWER
    Post-VOA Total Power
    OIF register 0xBBF8: Current Post-VOA Total Power
    We currently use a vendor defined 0x9000 register

TAI_NETWORK_INTERFACE_ATTR_CURRENT_PROVISIONED_CHANNEL_POWER
    Pre-VOA Channel/Signal Power
    OIF register 0xBBF4: Current Provisioned Channel Power

TAI_NETWORK_INTERFACE_ATTR_CURRENT_POST_VOA_PROVISIONED_CHANNEL_POWER
    Post-VOA Channel/Signal Power
    We currently use a vendor defined 0x9000 register

Comment (JumpeiHongo) [reply inc/tainetworkif.h]: Thank you for your information. I see. Then this is good for me.
We are better to confirm by using this kind of information to understand if possible.

Comment (ishidawataru) [reply inc/tainetworkif.h]: @samisaab Can you update the description of all of these input power attributes as you commented?

We've also started putting a reference register if it exists.
https://github.com/Telecominfraproject/oopt-tai/blob/master/inc/tainetworkif.h#L470

File: inc/tainetworkif.h [hunk 3]
```diff
@@ -511,6 +533,38 @@ typedef enum _tai_network_interface_attr_t
...
+     * @brief The RX power low alarm threshold in dBm
+     *
+     * @type #tai_float_t
+     * @flags READ_ONLY
+     */
+    TAI_NETWORK_INTERFACE_ATTR_RX_POWER_LOW_ALARM_THRESHOLD,
+
+    /**
+     * @brief The module general status
+     *
```
Comment (ishidawataru) [inline inc/tainetworkif.h]: Please add a `@reference` tag here

**Comment by ishidawataru** [comment]
@samisaab Sorry for the delay. Can you squash the commits to one? Then I'll merge this PR.

**Comment by samisaab** [comment]
@ishidawataru New attribute TAI_NETWORK_INTERFACE_ATTR_ALARM_NOTIFICATION is in location of TAI_NETWORK_INTERFACE_ATTR_CURRENT_POST_VOA_PROVISIONED_CHANNEL_POWER.  Which order should we pick to resolve conflict?

**Comment by ishidawataru** [comment]
@samisaab `TAI_NETWORK_INTERFACE_ATTR_ALARM_NOTIFICATION` is already merged. Please change the value of `TAI_NETWORK_INTERFACE_ATTR_CURRENT_POST_VOA_PROVISIONED_CHANNEL_POWER` and resolve the conflict.

**Comment by samisaab** [comment]
Will that break IPI's TAI interface later?

**Comment by ishidawataru** [comment]
As long as they use the right(latest) header files of TAI, it shouldn't be a problem.

**Comment by samisaab** [comment]
OK thanks.

**Comment by ishidawataru** [comment]
@samisaab Thanks!

**Pull request merged**